sdks/typescript/boot.go

```diff
@@ -87,7 +90,71 @@ func main() {
 		log.Fatalf("Failed to convert pipeline options: %v", err)
 	}
 
-	// (2) Invoke the Node entrypoint, passing the Fn API container contract info as flags.
+	// (2) Retrieve and install the staged packages.
+
+	dir := filepath.Join(*semiPersistDir, *id, "staged")
+	artifacts, err := artifact.Materialize(ctx, *artifactEndpoint, info.GetDependencies(), info.GetRetrievalToken(), dir)
+	if err != nil {
+		log.Fatalf("Failed to retrieve staged files: %v", err)
+	}
+
+	// Create a package.json that names given dependencies as overrides.
+	npmOverrides := make(map[string]string)
+	for _, v := range artifacts {
+		name, _ := artifact.MustExtractFilePayload(v)
+		path := filepath.Join(dir, name)
+		if v.RoleUrn == "beam:artifact:type:npm_dep:v1" {
+			// Npm cannot handle arbitrary suffixes.
+			suffixedPath := path + ".tar"
+			e := os.Rename(path, suffixedPath)
+			if e != nil {
+				log.Fatal(e)
+			}
+			npmOverrides[string(v.RolePayload)] = suffixedPath
+		}
+	}
+	if len(npmOverrides) > 0 {
```

Comment:
I don't quite follow why we're doing this - why can't we just install these dependencies with `npm install`?